src/api/src/helper.rs

```diff
@@ -707,7 +707,6 @@ pub fn pb_values_to_vector_ref(data_type: &ConcreteDataType, values: Values) ->
 }
 
 pub fn pb_values_to_values(data_type: &ConcreteDataType, values: Values) -> Vec<Value> {
-    // TODO(fys): use macros to optimize code
```

Comment:
The following code is stable and not quite in the same pattern.

We may not need to use macros here.